tasks/main.yml

```diff
@@ -54,29 +54,60 @@
   register: _gridinit_conf
   tags: configure
 
-- name: Generate {{ openio_redis_type }} configuration
-  lineinfile:
-    create: true
-    dest: "{{ openio_redis_sysconfig_dir }}/\
-      {{ openio_redis_servicename }}/{{ openio_redis_type }}.conf"
-    line: "{{ line }}"
-    mode: 0644
-    owner: openio
-    group: openio
-  with_items: "{{ lookup('template', openio_redis_type ~ '.conf.j2').splitlines() }}"
-  loop_control:
-    loop_var: line
-  register: _redis_conf
-  tags: configure
+- block:
+    - name: Generate {{ openio_redis_type }} configuration
+      lineinfile:
```

Comment:
Why is this not a standard template ?

Reply:
I can't use a `template`. The information about replication would be lost !

Reply:
For the "same file" version (no include), we should probably still:
- write before what sentinel writes by using `insertbefore`, *ie* before the line `# Generated by CONFIG REWRITE` (or better, a custom line of our own)
- use `regex` to match against directives, instead of trying to match the whole lines, *eg* match `sentinel monitor ` instead of `sentinel monitor YOLOREM-master-1 172.16.0.3 6011 2`

Reply:
- `insertbefore`: The file isn't valid (the run idempo duplicates the conf)
```
[root@2dd432efa7be redis-0]# cat redis.conf
# Ansible managed
daemonize no
pidfile "/run/oio/sds/TRAVIS/redis-0.pid"
port 6011
tcp-backlog 511
bind 172.17.0.2
timeout 0
tcp-keepalive 0
loglevel notice
logfile "/var/log/oio/sds/TRAVIS/redis-0/redis-0.log"
databases 16
save 900 1
save 300 10
save 60 10000
stop-writes-on-bgsave-error yes
rdbcompression yes
rdbchecksum yes
dbfilename dump.rdb
dir "/var/lib/oio/sds/TRAVIS/redis-0"
slave-serve-stale-data yes
slave-read-only yes
slave-priority 100
repl-diskless-sync no
repl-diskless-sync-delay 5
repl-disable-tcp-nodelay no
appendonly no
appendfilename "appendonly.aof"
appendfsync everysec
no-appendfsync-on-rewrite no
auto-aof-rewrite-percentage 100
auto-aof-rewrite-min-size 64mb
aof-load-truncated yes
lua-time-limit 5000
slowlog-log-slower-than 10000
slowlog-max-len 128
latency-monitor-threshold 0
notify-keyspace-events ""
hash-max-ziplist-entries 512
hash-max-ziplist-value 64
list-max-ziplist-entries 512
list-max-ziplist-value 64
set-max-intset-entries 512
zset-max-ziplist-entries 128
zset-max-ziplist-value 64
hll-sparse-max-bytes 3000
activerehashing yes
client-output-buffer-limit normal 0 0 0
client-output-buffer-limit slave 256mb 64mb 60
client-output-buffer-limit pubsub 32mb 8mb 60
hz 10
aof-rewrite-incremental-fsync yes
maxclients 10000
# Ansible managed
daemonize no
pidfile "/run/oio/sds/TRAVIS/redis-0.pid"
port 6011
tcp-backlog 511
bind 172.17.0.2
timeout 0
tcp-keepalive 0
loglevel notice
logfile "/var/log/oio/sds/TRAVIS/redis-0/redis-0.log"
databases 16
save 900 1
save 300 10
save 60 10000
stop-writes-on-bgsave-error yes
rdbcompression yes
rdbchecksum yes
dbfilename dump.rdb
dir "/var/lib/oio/sds/TRAVIS/redis-0"
slave-serve-stale-data yes
slave-read-only yes
slave-priority 100
repl-diskless-sync no
repl-diskless-sync-delay 5
repl-disable-tcp-nodelay no
appendonly no
appendfilename "appendonly.aof"
appendfsync everysec
no-appendfsync-on-rewrite no
auto-aof-rewrite-percentage 100
auto-aof-rewrite-min-size 64mb
aof-load-truncated yes
lua-time-limit 5000
slowlog-log-slower-than 10000
slowlog-max-len 128
latency-monitor-threshold 0
notify-keyspace-events ""
hash-max-ziplist-entries 512
hash-max-ziplist-value 64
list-max-ziplist-entries 512
list-max-ziplist-value 64
set-max-intset-entries 512
zset-max-ziplist-entries 128
zset-max-ziplist-value 64
hll-sparse-max-bytes 3000
activerehashing yes
client-output-buffer-limit normal 0 0 0
client-output-buffer-limit slave 256mb 64mb 60
client-output-buffer-limit pubsub 32mb 8mb 60
hz 10
aof-rewrite-incremental-fsync yes
maxclients 10000
maxmemory 0
maxmemory 0
```

```
[root@2dd432efa7be redissentinel-0]# cat redissentinel.conf
# Ansible managed
bind 172.17.0.2
port 6012
dir "/var/lib/oio/sds/TRAVIS/redissentinel-0"
daemonize no
pidfile "/run/oio/sds/TRAVIS/redissentinel-0.pid"
logfile "/var/log/oio/sds/TRAVIS/redissentinel-0/redissentinel-0.log"
sentinel myid 4064416682b763ba8fda4eb0366e975640080655
sentinel monitor TRAVIS-master-1 172.17.0.2 6011 2
min-slaves-to-write 1
min-slaves-max-lag 10
# Ansible managed
bind 172.17.0.2
port 6012
dir "/var/lib/oio/sds/TRAVIS/redissentinel-0"
daemonize no
pidfile "/run/oio/sds/TRAVIS/redissentinel-0.pid"
logfile "/var/log/oio/sds/TRAVIS/redissentinel-0/redissentinel-0.log"
sentinel monitor TRAVIS-master-1 172.17.0.2 6011 2
sentinel down-after-milliseconds TRAVIS-master-1 1000
min-slaves-to-write 1
min-slaves-max-lag 10
# Generated by CONFIG REWRITE
sentinel down-after-milliseconds TRAVIS-master-1 1000
sentinel config-epoch TRAVIS-master-1 0
sentinel leader-epoch TRAVIS-master-1 0
sentinel current-epoch 0
```

- `regex`  impossible to know if a line have 1 or n unique keyword. (example `save` and `sentinel *`